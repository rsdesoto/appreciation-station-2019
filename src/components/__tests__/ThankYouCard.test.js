import React from "react";
import { shallow } from "enzyme";
import ThankYouCard from "../ThankYouCard";

let testMember = {
  name: "Test mentor",
  isMentor: true
};

let testMessage = {
  text: "Thank you for being a great mentor, {NAME}!",
  isMentor: true
};

let testBgdImage = {
  file: {
    url: "testUrl"
  },
  maxTextWidth: 0.75,
  textPosition: "topLeft"
};

let subject;
let renderPrevArrow = jest.fn(); // mock functions - sets something just to test
let renderNextArrow = jest.fn();


describe("ThankYouCard", () => {

  describe("#render", () => {
    it("matches snapshot", () => {
      // the first time you render a component, it takes a snapshot. if nothing changed in DOM, these match
      // shallow render API - just use the "shallow" function
      subject = shallow(<ThankYouCard member={testMember}
                          message={testMessage}
                          backgroundImage={testBgdImage}
                          prevArrow={renderPrevArrow}
                          nextArrow={renderNextArrow}
                        />)
      expect(subject).toMatchSnapshot();
    })
  })

})







