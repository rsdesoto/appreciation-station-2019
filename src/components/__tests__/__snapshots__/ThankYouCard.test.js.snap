// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThankYouCard #render matches snapshot 1`] = `
<Fragment>
  <div
    className="ThankYouCardContainer"
  >
    <div
      className="ThankYouCard"
    >
      <canvas
        aria-label="Thank you for being a great mentor, Test mentor!"
        role="img"
      />
    </div>
  </div>
  <div>
    <a
      className="Share Button"
      download="Thank you Test mentor!.png"
      href="#"
      onClick={[Function]}
    >
      Download
    </a>
  </div>
</Fragment>
`;
